src/stability_sdk/client.py

```diff
@@ -314,30 +315,73 @@ def generate(
             parameters=[generation.StepParameter(**step_parameters)],
         )
 
-
         return self.emit_request(prompt=prompts, image_parameters=image_parameters)
     
     def upscale(
         self,
         init_image: Image.Image,
         height: int = None,
         width: int = None,
+        prompt: Union[str, generation.Prompt] = None,
+        steps: Optional[int] = 20,
+        cfg_scale: int = 7,
+        seed: Union[Sequence[int], int] = 0
     ) -> Generator[generation.Answer, None, None]:
+        """
+        Upscale an image.
+
+        :param init_image: Image to upscale.
+
+        Optional parameters for upscale method:
+
+        :param height: Height of the output images.
+        :param width: Width of the output images.
+        :param prompt: Prompt used in text conditioned models
+        :param steps: Number of diffusion steps
+        :param cfg_scale: Intensity of the prompt, when a prompt is used
+        :param seed: Seed for the random number generator.
+
+        Some variables are not used for specific engines, but are included for consistency.
+
+        Variables ignored in ESRGAN engines: prompt, steps, cfg_scale, seed
+
+        :return: Tuple of (prompts, image_parameters)
+        """
+        if isinstance(seed, int):
+            seed = [seed]
+        else:
+            seed = list(seed)
```

Comment:
Could remove the list of seeds handling and just have parameter to method be single int since the engine just uses the first seed and doesn't process multiple samples AFAIK

Reply:
Good point. I just changed the call to the upscale in the sdk, but seed still needs to be a list in grpc (generation.ImageParameters seed parameter)

Reply:
Cool yeah that's exactly what I meant (the method parameters accepting single seed instead of list)